File: Jav/programmers/src/level1/크레인_인형뽑기_게임.java
package level1;

import java.util.ArrayList;

class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        int N = board.length;
        ArrayList bag = new ArrayList();
        
        for (int move: moves){
            for (int i=0; i<N; i++){
                
                if (board[i][move-1] != 0){
                    bag.add(board[i][move-1]);
                    board[i][move-1] = 0;
                    
                    if (bag.size() >= 2){
                        if (bag.get(bag.size()-1) == bag.get(bag.size()-2)){
                            bag.remove(bag.size()-1);
                            bag.remove(bag.size()-1);
                            answer += 2;
                        }
                    }
                    break;
                }
            }
        }
        
        return answer;
    }
}

File: Programmers/level1/크레인_인형뽑기_게임.py
def solution(board, moves):
    answer = 0
    N = len(board)
    bag = []
    for move in moves:
        for i in range(N):
            if board[i][move-1]:
                print(i, move-1)
                bag.append(board[i][move-1])
                board[i][move-1] = 0
                if len(bag) >= 2:
                    if bag[-1] == bag[-2]:
                        bag.pop()
                        bag.pop()
                        answer += 2
                break
    return answer